swim/test_utils.go

```diff
@@ -121,6 +123,52 @@ func bootstrapNodes(t *testing.T, testNodes ...*testNode) {
 		})
 		require.NoError(t, err, "node must bootstrap successfully")
 	}
+	return hostports
+}
+
+func waitForConvergence(t *testing.T, timeout time.Duration, testNodes ...*testNode) {
+	var wg sync.WaitGroup
+	for _, tn := range testNodes {
+		wg.Add(1)
+		// execute the Protocol period on all nodes until the dissemination list is exhausted
+		go func(tn *testNode) {
+			for len(tn.node.disseminator.changes) > 0 {
+				tn.node.gossip.ProtocolPeriod()
+			}
+			wg.Done()
+		}(tn)
+	}
+
+	finished := make(chan bool, 1)
+	go func() {
+		wg.Wait()
+		finished <- true
+	}()
+
+	select {
+	case <-finished:
+		var checksums []uint32
```

Comment:
Would love to, but on is iterating over a `[]swim.Node` and the other iterates over `[]testNode` which are different types. I can DRY the loop that checks that all `[]uint32` to be the same, but that is all about it I think. Any thoughts?